sdk/communication/azure-communication-rooms/src/main/java/com/azure/communication/rooms/RoomsAsyncClient.java

```diff
@@ -219,159 +222,160 @@ Mono<Response<CommunicationRoom>> getRoomWithResponse(String roomId, Context con
-    public Mono<ParticipantsCollection> updateParticipants(String roomId, List<RoomParticipant> participants) {
-        return updateParticipants(roomId, participants, null);
+    public Mono<Object> upsertParticipants(String roomId, List<RoomParticipant> participants) {
+        return upsertParticipants(roomId, participants, null);
     }
 
-    Mono<ParticipantsCollection> updateParticipants(String roomId, List<RoomParticipant> participants, Context context) {
+    Mono<Object> upsertParticipants(String roomId, List<RoomParticipant> participants, Context context) {
         context = context == null ? Context.NONE : context;
         try {
             Objects.requireNonNull(participants, "'participants' cannot be null.");
             Objects.requireNonNull(roomId, "'roomId' cannot be null.");
-            return this.roomsClient
-            .updateParticipantsWithResponseAsync(roomId, toUpdateParticipantsRequest(participants), context)
-            .map(result -> new SimpleResponse<ParticipantsCollection>(
-                result, ParticipantsCollectionConverter.convert(result.getValue())))
-            .flatMap((Response<ParticipantsCollection> response) -> {
-                return Mono.just(response.getValue());
+
+            Map<String, ParticipantProperties> participantMap = ConvertRoomParticipantsToMapForUpsert(participants);
+            UpdateParticipantsRequest updateRequest = new UpdateParticipantsRequest().setParticipants(participantMap);
+
+            return this.participantsClient
+            .updateAsync(roomId, updateRequest)
+            .flatMap((Response<Object> response) -> {
+                return Mono.just(response);
             });
         } catch (RuntimeException ex) {
             return monoError(logger, ex);
         }
     }
 
     /**
-     * Update participants to an existing room with response.
+     * Upsert participants to an existing Room with response
      *
      * @param roomId The room id.
-     * @param participants The participant list.
-     * @return response for a successful update participants request.
+     * @param participants The participants list.
+     * @return response for a successful upsert participants room request.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<ParticipantsCollection>> updateParticipantsWithResponse(String roomId, List<RoomParticipant> participants) {
-        return updateParticipantsWithResponse(roomId, participants, null);
+    public Mono<Response<Object>> upsertParticipantsWithResponse(String roomId, List<RoomParticipant> participants) {
+        return upsertParticipantsWithResponse(roomId, participants, null);
     }
 
-    Mono<Response<ParticipantsCollection>> updateParticipantsWithResponse(String roomId, List<RoomParticipant> participants, Context context) {
+    Mono<Response<Object>> upsertParticipantsWithResponse(String roomId, List<RoomParticipant> participants, Context context) {
```

Comment:
We want to create an empty model called `UpsertParticipantsResult` and use that as return type